backend/src/backend/db/models.py

```diff
@@ -0,0 +1,129 @@
+from datetime import datetime
+from ipaddress import IPv4Address
+from typing import List, Optional
+from uuid import UUID
+
+from sqlalchemy import DateTime, Enum, ForeignKey, UniqueConstraint, text
+from sqlalchemy.dialects.postgresql import ARRAY, CITEXT, INET
+from sqlalchemy.orm import (
+    DeclarativeBase,
+    Mapped,
+    MappedAsDataclass,
+    mapped_column,
+    relationship,
+)
+from sqlalchemy.sql.schema import MetaData
+
+from backend.enums import StatusEnum
+
+
+class Base(MappedAsDataclass, DeclarativeBase):
+    # This map details the specific transformation of types between Python and
+    # PostgreSQL. This is only needed for the case where a specific PostgreSQL
+    # type has to be used.
+
+    type_annotation_map = {  # noqa: RUF012
+        str: CITEXT,  # transform Python str to PostgreSQL CITEXT
+        List[str]: ARRAY(
+            item_type=CITEXT
+        ),  # transform Python List[str] into PostgreSQL Array of strings
+        datetime: DateTime(
+            timezone=False
+        ),  # transform Python datetime into PostgreSQL Datetime without timezone
+        IPv4Address: INET,  # transform Python IPV4Address into PostgreSQL INET
+    }
+
+    # This metadata specifies some naming conventions that will be used by
+    # alembic to generate constraints names (indexes, unique constraints, ...)
+    metadata = MetaData(
+        naming_convention={
+            "ix": "ix_%(column_0_label)s",
+            "uq": "uq_%(table_name)s_%(column_0_name)s",
+            "ck": "ck_%(table_name)s_%(constraint_name)s",
+            "fk": "fk_%(table_name)s_%(column_0_name)s_%(referred_table_name)s",
+            "pk": "pk_%(table_name)s",
+        }
+    )
+    pass
+
+
+class Country(Base):
+    __tablename__ = "country"
+
+    code: Mapped[str] = mapped_column(primary_key=True)
+    name: Mapped[str]
+
+    worker_id: Mapped[Optional[UUID]] = mapped_column(
+        ForeignKey("worker.id"), init=False
+    )
+    worker: Mapped[Optional["Worker"]] = relationship(
+        back_populates="countries", init=False
+    )
+    mirrors: Mapped[List["Mirror"]] = relationship(
+        back_populates="country",
+        init=False,
+        cascade="all, delete-orphan",
+    )
+
+    __table_args__ = (UniqueConstraint("name", "code"),)
+
+
+class Mirror(Base):
+    __tablename__ = "mirror"
+
+    base_url: Mapped[str] = mapped_column(primary_key=True)
```

Comment:
We'll want to use the ID as PKEY since the base_url is not under our control and subject to change. Given this is indexed, makes no sense to use full URL over ID (netloc as suggested in ticket)

Reply:
By `id`, do you mean the MirrorBrain identifier or a database-generated ID like in other fields? The mirrobrain identifier is not accessible from [the mirror list](https://saimei.ftp.acc.umu.se/mirror/kiwix.org/mirrors.html). I had added an `id` field under the comment which begins the listing of metadata obtained from MB. I guess I would rename it explicitly as `identifier` or should I remove it entirely?

Reply:
I indicated in the ticket we'd use the hostname of the mirror's base_url as ID.